pkg/backend/httpstate/backend.go

```diff
@@ -1177,18 +1178,47 @@ func (b *cloudBackend) ExportDeploymentForVersion(
 		return nil, errors.Errorf("%q is not a valid stack version. It should be a positive integer.", version)
 	}
 
-	return b.exportDeployment(ctx, stack.Ref(), &versionNumber)
+	return b.exportDeployment(ctx, stack, &versionNumber, showSecrets)
 }
 
 // exportDeployment exports the checkpoint file for a stack, optionally getting a previous version.
 func (b *cloudBackend) exportDeployment(
-	ctx context.Context, stackRef backend.StackReference, version *int) (*apitype.UntypedDeployment, error) {
+	ctx context.Context, stk backend.Stack, version *int, showSecrets bool) (*apitype.UntypedDeployment, error) {
```

Comment:
This is just in the cloud backend.  Do we have the ability to do this higher in the stack so that we don't have to implement separately for both backends?  It feels like at least in theory this shouldn't have anything to do with the backend.

Reply:
I have not investigated this further up the stack - I was trying to be as close to the code as possible - we can of course try and move it further up but it could be more disruptive